app/scripts/directives/keyValueEditor.js

```diff
@@ -286,6 +302,36 @@
                 }
               });
 
+              var findReferenceValue = function(entry) {
```

Comment:
Think its possible to break this one down a bit?  There are a lot of `return`s at various levels.  Might be helpful to clarify (via function names) what is going on?